docs/adr/ADR-009-structured-json-logging.md

```diff
@@ -0,0 +1,399 @@
+# ADR-009: Structured JSON Logging for Production Systems
+
+**Status:** Accepted
+**Date:** 2025-11-14
+**Authors:** cassing
+**Supersedes:** N/A
+**Superseded by:** N/A
+
+## Context
+
+Production systems need machine-readable logs for effective monitoring, debugging, and alerting. Traditional text-based logs are difficult to parse and query in log aggregation systems like Elasticsearch, Splunk, or CloudWatch.
+
+**Current Situation:**
+
+- API services generate thousands of log entries per hour
+- Logs need to be searchable by specific fields (request_id, workspace, usergroup, etc.)
+- Debugging requires correlating logs across multiple requests and workers
+- Different environments have different logging needs (production vs development)
+
+**Problems with Plain Text Logging:**
+
+- **No structured queries:** Can't easily filter by specific fields
+- **Regex parsing required:** Fragile and error-prone in log aggregation systems
+- **Poor correlation:** Hard to trace all logs for a single request
+- **Lost context:** Extra fields (workspace, usergroup, action_type) not easily accessible
+- **No metric extraction:** Can't easily extract numeric values (duration, status codes)
+
+**Requirements:**
+
+- Machine-readable JSON logs for production (log aggregation systems)
+- Human-readable logs for development (debugging with stacktraces)
+- Automatic request tracking across all log entries
+- Support for arbitrary extra fields (workspace, usergroup, action_type, etc.)
+- Easy to search, filter, and query in log aggregation tools
+- No performance impact on request processing
+
+**Constraints:**
+
+- Must work with Python's standard logging library
+- Must integrate with FastAPI middleware
+- Must support both synchronous and asynchronous code
+- Environment variable configuration (no code changes)
+
+## Decision
+
+We adopt **dual-mode structured logging** with JSON format for production and standard format for development.
+
+**Production Mode (JSON):**
+
+- Use `python-json-logger` library for structured JSON output
```

Comment:
have you tried https://www.structlog.org/en/stable/? seems a more popular one than `python-json-logger`

Reply:
structlog looks promising. I've created https://issues.redhat.com/browse/APPSRE-13018 to tackle that later